textworld/generator/inform7/world2inform7.py

```diff
@@ -731,8 +744,18 @@ def gen_source(self, seed: int = 1234) -> str:
                 say "Can't see any [object] on the floor! Try taking the [object] from the [container] instead.";
                 rule fails.
 
-        """)
+        Understand "take [something]" as removing it from.
+
+        Rule for supplying a missing second noun while removing:
+            if restrict commands option is false and noun is on a supporter (called the supporter):
+                now the second noun is the supporter;
+            else if restrict commands option is false and noun is in a container (called the container):
+                now the second noun is the container;
+            else:
+                try taking the noun;
+                say "";
```

Comment:
Is this necessary?

Reply:
Yes, I will add a comment about it. `say` triggers a change of a global variable, otherwise some default text is printed (which I don't want). The counterpart that trick is that a newline is printed but it's not a big deal.

Reply:
Done + I rebased on top of master.